tests/sentry/ratelimits/test_redis_concurrent.py

```diff
@@ -0,0 +1,94 @@
+import time
+import uuid
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime, timedelta
+
+from freezegun import freeze_time
+
+from sentry.ratelimits.concurrent import DEFAULT_MAX_TTL_SECONDS, ConcurrentRateLimiter
+from sentry.testutils import TestCase
+
+
+class ConcurrentLimiterTest(TestCase):
+    def setUp(self):
+        self.backend = ConcurrentRateLimiter()
+
+    def test_add_and_remove(self):
+        limit = 8
+        with freeze_time("2000-01-01"):
+            for i in range(1, limit + 1):
+                assert (
+                    self.backend.start_request("foo", limit, f"request_id{i}").current_executions
+                    == i
+                )
+
+            info = self.backend.start_request("foo", limit, "request_id_12")
```

Comment:
Is the `request_id` number 12 just because that's higher than the `limit` and you want it to be over the limit?

Reply:
correct. I added a comment. let me know if that's more clear